src/HttpAsyncClientEmulator.php

```diff
@@ -29,7 +28,7 @@ public function sendAsyncRequest(RequestInterface $request)
     {
         try {
             return new Promise\FulfilledPromise($this->sendRequest($request));
-        } catch (Exception $e) {
+        } catch (\Exception $e) {
```

Comment:
The RejectedPromise() takes a \Exception in its constructor. That's why we should use \Exception here.